pkg/daemon/daemon.go

```diff
@@ -471,32 +472,29 @@ func ReexecuteForTargetRoot(target string) error {
 		}
 	}
 
-	// Be idempotent
-	targetBinExist, err := fileExists(targetBin)
+	f, err := renameio.TempFile(targetBinDir, targetBin)
```

Comment:
The big difference here is we're not calling `os.Create` on a possibly-executing binary.  Instead we're doing:

- Create /run/bin/machine-config-daemon.temp
- Write to it
- rename(/run/bin/machine-config-daemon.temp, /run/bin/machine-config-daemon)

Which is really the pattern we should use to write *all* files in general, and it will avoid the "Text file busy" if the binary is currently running.